src/Confluent.Kafka/Consumer.cs

```diff
@@ -1053,376 +741,414 @@ public void Assign(IEnumerable<TopicPartitionOffset> partitions)
         ///     have been committed yet.
         /// </param>
         public void Assign(IEnumerable<TopicPartition> partitions)
-            => kafkaHandle.Assign(partitions.Select(p => new TopicPartitionOffset(p, Offset.Invalid)).ToList());
-
-        /// <summary>
-        ///     Stop consumption and remove the current topic/partition assignment.
-        /// </summary>
-        public void Unassign()
-            => kafkaHandle.Assign(null);
-
-        /// <summary>
-        ///     Poll for new messages / consumer events. Blocks until a new 
-        ///     message or event is ready to be handled or the timeout period
-        ///     <paramref name="millisecondsTimeout" /> has elapsed.
-        /// </summary>
-        /// <param name="message">
-        ///     A consumed message, or null if no messages are 
-        ///     available for consumption.
-        /// </param>
-        /// <param name="millisecondsTimeout">
-        ///     The maximum time to block (in milliseconds), or -1 to 
-        ///     block indefinitely. You should typically use a
-        ///     relatively short timout period because this operation
-        ///     cannot be cancelled.
-        /// </param>
-        /// <returns>
-        ///     true: a message (with non-error state) was consumed.
-        ///     false: no message was available for consumption.
-        /// </returns>
-        /// <remarks>
-        ///     Will invoke events for OnPartitionsAssigned/Revoked,
-        ///     OnOffsetsCommitted, OnConsumeError etc. on the calling 
-        ///     thread.
-        /// </remarks>
-        public bool Consume(out Message message, int millisecondsTimeout)
         {
-            if (kafkaHandle.ConsumerPoll(out message, (IntPtr)millisecondsTimeout))
-            {
-                switch (message.Error.Code)
-                {
-                    case ErrorCode.NoError:
-                        return true;
-                    case ErrorCode.Local_PartitionEOF:
-                        OnPartitionEOF?.Invoke(this, message.TopicPartitionOffset);
-                        return false;
-                    default:
-                        OnConsumeError?.Invoke(this, message);
-                        return false;
-                }
-            }
-
-            return false;
+            assignCalled = true;
+            kafkaHandle.Assign(partitions.Select(p => new TopicPartitionOffset(p, Offset.Invalid)).ToList());
```

Comment:
we might want to make assignCalled a counter and then warn, or throw, if assign/unassign was called multiple times from an OnPartitiosn.. handler list execution, since that is a bug.

Reply:
great idea